### ISSUE TITLE
Do a version 1.1.0 release

### ISSUE THREAD

**Issue by edwardhartnett** [NOAA-EMC/hpc-stack#72]
Looks like we need a versioned release of hpc-stack.

**Comment by kgerheiser** [comment]
Yes, this is a good idea.

I've been meaning to make an issue.

Need to tag wgrib2 and update it here. The current version has a serious bug with the executable.

Maybe update ESMF?

~~Update the tag of g2c so it can build with GCC 10~~

I meant grib_util

UPP 10.0?

**Comment by edwardhartnett** [comment]
@Hang-Lei-NOAA is there already a tag to a working hpc-stack that we want to release?

Or do we want to release the develop branch?

**Comment by edwardhartnett** [comment]
@kgerheiser we cannot just update tags. If we want to update it, we need to do a g2c release as well...

**Comment by kgerheiser** [comment]
~~g2c has an updated release from a few days ago which contains the fix~~

wgrib2 has been updated with its bugfix. Just need to tag and make a release.

They've been using a newer version of ESMF already, just update the tag. (not sure if this is needed)

UPP 10.0 is also tagged.

**Comment by Hang-Lei-NOAA** [comment]
Great, Kyle please tag wgrib2 and I will update in the test versions.

Other changes have been added to the test version

**Comment by kgerheiser** [comment]
Well, wgrib2 is the main thing I wanted to update, which I've done.

I said g2c when I meant grib_util. 


I mention ESMF because the model has been testing with a different beta version while our release version has beta snapshot 27.

**Comment by Hang-Lei-NOAA** [comment]
I will update grub_util today as well. Thanks

**Comment by Hang-Lei-NOAA** [comment]
@kgerheiser I saw the grib_util is still the tag v1.2.1.  Is this the new tag you mean?
@edwardhartnett I think that they are using the hpc-stack-nco for test now. the new release should bring this back to the develop and releasse

**Comment by kgerheiser** [comment]
No, I was wrong about there being a tag for it already. It was merged into develop, but there's no tag. We can either just leave it as is (only affects GCC 10), or we would have to create a release for grib_util.

**Comment by Hang-Lei-NOAA** [comment]
I strongly recommend that we have a new tag. We can contact the code
manager to see if he agrees with this. @Boi Vuong - NOAA Federal
<boi.vuong@noaa.gov>

**Comment by BoiVuong-NOAA** [comment]
Kyle and Hang,
I already delivered code (grib_util v1.2.0) to NCO.  We can create a new tag for it or leave it as it in the branch develop.
NOTE: NCO will not accept grib_util build with dynamic libraries on DELL (MARS and VENUS) and CRAY (LUNA and SURGE).
Do you want a new tag for grib_util ?

**Comment by kgerheiser** [comment]
There's no affect on NCO or what you already delivered

**Comment by BoiVuong-NOAA** [comment]
I think we do not need a new tag for it, agree ?  Let me know.

**Comment by Hang-Lei-NOAA** [comment]
I am Okay with all decisions

**Comment by kgerheiser** [comment]
We'll need a new tag for this commit off of the develop branch:

 https://github.com/NOAA-EMC/NCEPLIBS-grib_util/pull/44

Just need to increment the last number by 0.0.1 to 1.2.2 to indicate it's a bugfix.

I can create it if you don't mind.

**Comment by BoiVuong-NOAA** [comment]
Okay, Would you do it ?

**Comment by kgerheiser** [comment]
Created

https://github.com/NOAA-EMC/NCEPLIBS-grib_util/releases/tag/v1.2.2

**Comment by edwardhartnett** [comment]
I have drafted the release notes.

Other than grib_util and wgrib2, what other NCEPLIBS libraries got a new release in support of hpc-stack 1.1.0?

Other than ESMF, what other external libraries were updated?

What version of ESMF will hpc-stack 1.1.0 have?

**Comment by Hang-Lei-NOAA** [comment]
Above previous release,
g2c/1.6.2,
upp/10.0.0 to replace nceppost
HDF5 and NETCDF compiling options to match the GFSv16 requirements.
Add option of NCO_V to build only NCEPLIBS libraries by using machine
installed third party libraries on WCOSS machines.
Replace bash -> sh to let hpc-stack working on jet and WCOSS2
Add cheyenne build options.

**Comment by kgerheiser** [comment]
ip2 will also be updated to fix the conflicting module name with ip

**Comment by edwardhartnett** [comment]
Is prod_util being updated for the hpc-stack 1.1.0 release?

**Comment by edwardhartnett** [comment]
Here's what I have now. Is this a complete list?

## New This Release

### hpc-stack Updates

* HDF5 and NETCDF compiling options to match the GFSv16 requirements.
* Add option of NCO_V to build only NCEPLIBS libraries by using machine.
* Installed third party libraries on WCOSS machines.
* Replace bash -> sh to let hpc-stack working on jet and WCOSS2
* Add cheyenne build options.

### NCEPLIBS Updates

* Updated to grib_util-2.2.0 (see .https://github.com/NOAA-EMC/NCEPLIBS-grib_util).
* Updated to NCEPLIBS-wgrib2-v2.0.8-cmake-v6 (see https://github.com/NOAA-EMC/NCEPLIBS-wgrib2).
* Updated to NCEPLIBS-ip2-1.1.2 (see https://github.com/NOAA-EMC/NCEPLIBS-ip2).
* Updated to NCEPLIBS-g2c-1.6.2 (see https://github.com/NOAA-EMC/NCEPLIBS-g2c).

### External Package Updates
* Updated ESMF to bs36 (bs27 is also still supported) (see #77).
* Updated to UPP-10.0 (see #31).

**Comment by climbfuji** [comment]
Will the bash -> sh change break macOS? Missing is probably also the "Add gaea build options", depending on whether my PR is accepted or not.

**Comment by Hang-Lei-NOAA** [comment]
Thanks Edward.

**Comment by BoiVuong-NOAA** [comment]
Hi Ed,

Just a question. Why do we have grib_util version 2.2.0 ?  What are 
changes ?

Thanks,

Boi

**Comment by edwardhartnett** [comment]
@BoiVuong-NOAA see https://github.com/NOAA-EMC/hpc-stack/issues/67.

Basically to compile with gfortran 10.

**Comment by BoiVuong-NOAA** [comment]
I think the version should be 1.2.1 instead 2.2.0   Is is a typo ?

**Comment by edwardhartnett** [comment]
It was a typo. But 1.2.2 is the version of grib_util that was just released.

Here's what I have now:

## New This Release

### hpc-stack Updates

* HDF5 and NETCDF compiling options to match the GFSv16 requirements.
* Add option of NCO_V to build only NCEPLIBS libraries by using machine.
* Installed third party libraries on WCOSS machines.
* Replace bash -> sh to let hpc-stack working on jet and WCOSS2.
* Add cheyenne build options.

### NCEPLIBS Updates

* Updated to grib_util-1.2.2 (see .https://github.com/NOAA-EMC/NCEPLIBS-grib_util).
* Updated to NCEPLIBS-wgrib2-v2.0.8-cmake-v6 (see https://github.com/NOAA-EMC/NCEPLIBS-wgrib2).
* Updated to NCEPLIBS-ip2-1.1.3 (see https://github.com/NOAA-EMC/NCEPLIBS-ip2).
* Updated to NCEPLIBS-g2c-1.6.2 (see https://github.com/NOAA-EMC/NCEPLIBS-g2c).

### External Package Updates
* Updated ESMF to bs36 (bs27 is also still supported) (see #77).
* Updated to UPP-10.0 (see #31).

### Complete List of Changes

For all changes in this release see https://github.com/NOAA-EMC/hpc-stack/projects/1.

**Comment by edwardhartnett** [comment]
We are rapidly closing in on the release! I am not normally a fan of pre-releases, but GitHub allows them, and this project is a good example of where they can be useful.

Let's mark a 1.1.0 release on Monday, and mark it as a pre-release while it is installed and tested everywhere. When we have good news from all our platforms we will make it the release, which just changes how GitHub displays it - the tag remains the same. So we don't have to re-install after moving from pre-release to release.

**Comment by climbfuji** [comment]
> We are rapidly closing in on the release! I am not normally a fan of pre-releases, but GitHub allows them, and this project is a good example of where they can be useful.
> 
> Let's mark a 1.1.0 release on Monday, and mark it as a pre-release while it is installed and tested everywhere. When we have good news from all our platforms we will make it the release, which just changes how GitHub displays it - the tag remains the same. So we don't have to re-install after moving from pre-release to release.

That sounds like an excellent approach to me, and it will allow me to build the stack and the model (!) on mac again. I had to  skip the ufs-weather-model that required the now-deleted hpc-stack-nco branch.

**Comment by kgerheiser** [comment]
I think we're ready to give it a try now.

**Comment by edwardhartnett** [comment]
Awesome! I think after #99 is merged, you should do the release.

I will take one last look at the release notes on GitHub...

**Comment by kgerheiser** [comment]
I'm doing a test on Orion before I make the release. I built hpc-stack and am now building the weather model. Going to run the regression tests, and once those finish I'll create the release.

**Comment by kgerheiser** [comment]
I made the release 

https://github.com/NOAA-EMC/hpc-stack/releases/tag/v1.1.0

I have started installing on Jet and Orion.

**Comment by Hang-Lei-NOAA** [comment]
Good, Kyle. I have started installing it on wcoss dell.
And will port the pkg to Hera for further installation after dell
downloaded all.

**Comment by edwardhartnett** [comment]
I believe everything is done for the 1.1.0 release. I will close this issue and this release...
